keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/ConstraintEvaluator.kt

```diff
@@ -33,6 +33,12 @@ interface ConstraintEvaluator<T : Constraint> {
 
   val eventPublisher: ApplicationEventPublisher
 
+  /**
+   * returns true if a constraint should be run for every environment in every delivery config, without being
+   * exposed to delivery config author.
+   */
+  fun isImplicit(): Boolean = false
```

Comment:
How would this work for a constraint that was implicit but only applied to certain artifact types?

Reply:
It would run for every environment, but if that artifact wasn't there then it would just return true. If the artifact or thing you cared about was there, then you'd add logic to decide what to do in that case

Reply:
So, for example the "AMI exists" constraint should just return `true` for a Docker artifact?

Reply:
Yep! but in this case "true" is more like "ehh idk sure go ahead pal"

Reply:
Right, that's what I figured. Makes sense.